### ISSUE TITLE
Deprecation warnings during tests when shots==1

### ISSUE THREAD

**Issue by diego-plan9** [QISKit/qiskit-sdk-py#441]
After #435, there are a number of tests that result in `DeprecationWarnings` during `_validate()`:

```
DeprecationWarning: The behavior of getting statevector from simulators by setting shots=1 is deprecated and will be removed. Use the local_statevector_simulator instead, or place explicit snapshot instructions.
```

```
test/python/test_qasm_simulator_cpp.py::TestLocalQasmSimulatorCpp::test_conditionals
test/python/test_qasm_simulator_cpp.py::TestLocalQasmSimulatorCpp::test_qobj_reset
test/python/test_qasm_simulator_cpp.py::TestLocalQasmSimulatorCpp::test_qobj_save_load
test/python/test_qasm_simulator_cpp.py::TestLocalQasmSimulatorCpp::test_qobj_single_qubit_gates
test/python/test_qasm_simulator_cpp.py::TestLocalQasmSimulatorCpp::test_qobj_two_qubit_gates
test/python/test_qasm_simulator_py.py::TestLocalQasmSimulatorPyPy::test_qasm_simulator_single_shot
```

Can we update the tests accordingly so they do not produce such warnings?

Additionally, while looking at the `_validate()` methods:
* can we make the `_validate()` methods for the statevector simulators emit `logger.warning` instead of `logger.info` when the number of shots is different than 1? This would make the message visible to the user by default, and let them know the simulator is ignoring his value for `shots`  instead of silently dropping it, fitting better into the severity levels.
* is there the possibility that the user might be interested in running a program against a QASM simulator with just 1 shot (ie. not assuming always that it will imply that the user wants to use a statevector simulator)? 

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment

* Version used:
* Environment name and version (e.g. Python 3.6.1):
* Operating System and version:

**Comment by 1ucian0** [comment]
This will be address by @ajavadia #422

**Comment by ajavadia** [comment]
@diego-plan9  I submitted patch #458 for the first issue.
The second one is resolved in #422, which i think we can merge soon after the release.
The third issue: no, that's why we distinguished the qasm simulator from statevector simulator. With qasm, the assumption is you are getting some measurement result. So with one shot, you only get a random number, which isn't useful at all. You would want to do many shots to get output histograms. With statevector simulator, you do get the full probability distribution so one shot is enough.

**Comment by yaelbh** [comment]
Both qasm_simulator_py and qasm_simulator_projectq calculate a classical state if the number of shots is 1.
What is the plan for the classical state, now that we distinguish between qasm and statevector simulators?
Possibly what makes sense is that the qasm simulators will provide the option of saving the classical state for each snapshot. Then the statevector simulators will provide not only the state vector in the final snapshot but also the classical state.
What do you think?
